app/transcript.py

```
@@ -14,6 +15,76 @@
+    unspecified (0.0.0.0 / ::), and IANA-reserved ranges — catching all the
+    categories a pure denylist approach could miss.
+    """
+    import socket as _socket
+    import ipaddress as _ipaddress
+
+    # Capture the original *inside* the lock so we always snapshot the real
+    # getaddrinfo, never an already-patched version from another thread.
+    with _ssrf_lock:
+        _orig_getaddrinfo = _socket.getaddrinfo
+
+        def safe_getaddrinfo(host, port, family=0, type=0, proto=0, flags=0):
```

Ruff (0.15.17)

[error] Function argument `type` is shadowing a Python builtin

(A002)